os/arch/arm/src/common/up_createstack.c

```diff
@@ -180,12 +184,20 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		/* Use the kernel allocator if this is a kernel thread */
 
 		if (ttype == TCB_FLAG_TTYPE_KERNEL) {
-			tcb->stack_alloc_ptr = (uint32_t *)kmm_malloc(stack_size);
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION	/* address must be aligned to 32 byte for MPU region */
+			tcb->stack_alloc_ptr = (uint32_t *)kmm_memalign(MPU_REGION_ADDRESS_ALIGNMENT, stack_size);
```

Comment:
In description of MPU_STACK_OVERFLOW_PROTECTION, you say `This is applicable to user mode tasks created in protected build.`. Then, why is this necessary? This is for kernel thread.

Reply:
yes, removed the kernel thread stack creation changes